common/src/main/java/noobanidus/mods/lootr/common/block/entity/BlockEntityTicker.java

```diff
@@ -20,195 +20,130 @@
-            if (LootrAPI.isTaggedStructurePresent(level, thisPos, LootrTags.Structure.STRUCTURE_BLACKLIST, entry.getPosition())) {
-              toRemove.add(entry);
-              continue;
-            }
-          } else if (registry.getTag(LootrTags.Structure.STRUCTURE_WHITELIST).filter(tag -> tag.size() != 0).isPresent()) {
-            if (!LootrAPI.isTaggedStructurePresent(level, thisPos, LootrTags.Structure.STRUCTURE_WHITELIST, entry.getPosition())) {
-              toRemove.add(entry);
-              continue;
-            }
-          }
-        }
-
-        BlockEntity blockEntity = level.getBlockEntity(entry.getPosition());
-        if (!(blockEntity instanceof RandomizableContainerBlockEntity be) || LootrAPI.resolveBlockEntity(blockEntity) instanceof ILootrBlockEntity) {
-          toRemove.add(entry);
-          continue;
-        }
-        if (be.getLootTable() == null || LootrAPI.isLootTableBlacklisted(be.getLootTable())) {
-          toRemove.add(entry);
-          continue;
-        }
-        BlockState stateAt = level.getBlockState(entry.getPosition());
-        BlockState replacement = LootrAPI.replacementBlockState(stateAt);
-        if (replacement == null) {
-          toRemove.add(entry);
-          continue;
-        }
-        // Save specific data. Currently, this includes the LockCode (all platforms), along with NeoForge's getPersistentData.
-        DataToCopy data = PlatformAPI.copySpecificData(be);
-        ResourceKey<LootTable> table = be.getLootTable();
-        long seed = be.getLootTableSeed();
-        // IMPORTANT: Clear loot table to prevent loot drop when container is destroyed
-        be.setLootTable(null);
-        level.destroyBlock(entry.getPosition(), false);
-        level.setBlock(entry.getPosition(), replacement, 2);
-        BlockEntity newBlockEntity = level.getBlockEntity(entry.getPosition());
-        PlatformAPI.restoreSpecificData(data, newBlockEntity);
-        if (LootrAPI.resolveBlockEntity(newBlockEntity) instanceof ILootrBlockEntity && newBlockEntity instanceof RandomizableContainerBlockEntity rbe) {
-          rbe.setLootTable(table, seed);
-        } else {
-          LootrAPI.LOG.error("replacement " + replacement + " is not an ILootrBlockEntity " + entry.getDimension() + " at " + entry.getPosition());
-        }
+      ServerLevel level = server.getLevel(entry.getDimension());
+      ChunkPos chunkPos = entry.getChunkPos();
+      if (level == null || !LootrAPI.isWorldBorderSafe(level, chunkPos)) {
+        iterator.remove();
+        continue;
+      }
 
-        toRemove.add(entry);
+      ServerChunkCache chunkSource = level.getChunkSource();
+      if (!chunkSource.hasChunk(chunkPos.x, chunkPos.z)) {
+        // the chunk has unloaded. this entry is no longer valid, and it will be added again if the chunk loads again.
+        iterator.remove();
+        continue;
```

Comment:
I think this is an important new check, it keeps the list of `blockEntityEntries` much smaller than before, which improves performance. Without this, they would only be removed after "age" become large enough.